www/search

```diff
@@ -1458,11 +1458,16 @@ else if ($term || $browse)
                     $tagsMatched = array();
                     foreach ($specials as $s) {
                         if ($s[0][0] == "tags") {
+                            $result = mysqli_execute_query($db, 'select totag from tagsynonyms where fromtag = ?', [$s[1]]);
+                            $synonyms = array_merge(...mysqli_fetch_all($result, MYSQLI_NUM));
+                            $matchables = array_merge([$s[1]], $synonyms);
                             foreach ($tags as $t) {
-                                if (strpos(strtolower($t),
-                                           strtolower($s[1])) !== false) {
-                                    $tagsMatched[$t] = true;
-                                    break;
+                                foreach ($matchables as $m) {
```

Comment:
I don't know how much data this is supposed to have, but here's an optimization suggestion, in case it's useful:

Move the `strtolower` out of the nested loop. Compute it once for `$t` and `$m`. This will need a mapping from the lowercased strings to their original form. (Need to be careful of duplicates, like "aa" mapping to both "AA" and "Aa", but maybe this should be enforced when saving new synonyms.)

If you don't mind importing another package, you can even use [set intersection](https://www.php.net/manual/en/class.ds-set.php) instead of the loop.